### ISSUE TITLE
Build fails on Arch due to ncurses6 

### ISSUE THREAD

**Issue by dylanaraps** [karlstav/cava#70]
ncurses6 was added to the testing repos on arch and cava fails to build because it can't find ncurses5.

For ncurses6 users, I've found two temporary solutions: 
* Symlink the ncurses6 binary

        sudo ln -s /bin/ncursesw6-config /bin/ncursesw5-config

* Or Change "ncursesw5-config" on line 18 of the makefile to "ncursesw6-config" 

Rebuild after either symlinking or editing the makefile and it should build and run fine.

**Comment by karlstav** [comment]
The use of 'ncursesw5-config' was added in 10cde13 to make the ncurses build portabale. (the ncurses library had different names in ubuntu and arch (in arch ncursesw was bundled in ncurses while in ubuntu it was separate)). 

@dylanaraps great work on finding a work around, but I guess we need some kind of version detection. Maybe it's time to make a [configure script](https://en.wikipedia.org/wiki/Configure_script)?

**Comment by dylanaraps** [comment]
@karlstav thanks! A configure script would be great, ncmpcpp has one that detects the ncurses version and we could probably modify that. 

Relevant part of ncmpcpp's configure file
https://github.com/arybczak/ncmpcpp/blob/master/configure.ac
```
dnl ========================
dnl = checking for ncurses =
dnl ========================
if test "$unicode" = "yes" ; then
	curses_config_bin="ncursesw6-config ncursesw5-config"
	AC_DEFINE([NCMPCPP_UNICODE], [1], [enables unicode support])
else
	curses_config_bin="ncurses6-config ncurses5-config"
fi

AC_PATH_PROGS(CURSES_CONFIG, $curses_config_bin)
if test "$CURSES_CONFIG" != "" ; then
	CPPFLAGS="$CPPFLAGS `$CURSES_CONFIG --cflags`"
	LIBS="$LIBS `$CURSES_CONFIG --libs`"
fi
AC_CHECK_LIB(ncursesw, initscr,
	curses_lib=ncursesw,
	curses_lib=ncurses
)

AC_CHECK_LIB($curses_lib, initscr,
	if test "$CURSES_CONFIG" = "" ; then
		LIBS="$LIBS -l$curses_lib"
	fi
	,
	AC_MSG_ERROR([$curses_lib library is required])
	)
AC_CHECK_HEADERS([curses.h], , AC_MSG_ERROR([missing curses.h header]))
```

**Comment by karlstav** [comment]
I have never done config scripts before and I'm completely striped from time right now, but I created a  config-script branch in the hope that someone might feel like helping out.

This is the plan (as I see it):

need to have:
* automatic detection of ncurses version

nice to have:
* detect fftw, if no exit with some kind of install tips
* detect alsa dev files, build with mpd fifo support only if not installed, but throw some kind of warning and hint about how to install alsa dev files
* if no ncurses(w) are installed build with only "noncurses"-mode, but throw warnings as above

**Comment by karlstav** [comment]
fixed by config sctipr in 0.4.0
